source/draw/RasterRgb.ooc

```diff
@@ -0,0 +1,127 @@
+	}
+	draw: override func ~ImageSourceDestination (image: Image, source, destination: IntBox2D) {
+		rgb: RasterRgb = null
+		if (image instanceOf(RasterRgb))
+			rgb = image as RasterRgb
+		else if (image instanceOf(RasterImage))
+			rgb = RasterRgb convertFrom(image as RasterImage)
+		else
+			Debug error("Unsupported image type in RgbRasterCanvas draw")
+		this _resizePacked(rgb buffer pointer as ColorRgb*, rgb, source, destination)
+		if (rgb != image)
+			rgb referenceCount decrease()
+	}
+}
+
+RasterRgb: class extends RasterPacked {
+	bytesPerPixel ::= 3
+	init: func ~allocate (size: IntVector2D) { super~allocate(size) }
+	init: func ~allocateStride (size: IntVector2D, stride: UInt) { super(size, stride) }
+	init: func ~fromByteBufferStride (buffer: ByteBuffer, size: IntVector2D, stride: UInt) { super(buffer, size, stride) }
+	init: func ~fromByteBuffer (buffer: ByteBuffer, size: IntVector2D) { this init(buffer, size, this bytesPerPixel * size x) }
+	init: func ~fromRasterRgb (original: This) { super(original) }
+	init: func ~fromRasterImage (original: RasterImage) { super(original) }
+	create: override func (size: IntVector2D) -> Image { This new(size) }
+	copy: override func -> This { This new(this) }
+	apply: override func ~rgb (action: Func(ColorRgb)) {
+		for (row in 0 .. this size y)
+			for (pixel in 0 .. this size x) {
+				pointer := this buffer pointer + pixel * this bytesPerPixel + row * this stride
+				color := (pointer as ColorRgb*)@
+				action(color)
+			}
+	}
+	apply: override func ~bgr (action: Func(ColorBgr)) {
+		convert := ColorConvert fromRgb(action)
+		this apply(convert)
+		(convert as Closure) free()
+	}
+	apply: override func ~yuv (action: Func(ColorYuv)) {
+		convert := ColorConvert fromRgb(action)
+		this apply(convert)
+		(convert as Closure) free()
+	}
+	apply: override func ~monochrome (action: Func(ColorMonochrome)) {
+		convert := ColorConvert fromRgb(action)
+		this apply(convert)
+		(convert as Closure) free()
+	}
+	resizeTo: override func (size: IntVector2D) -> This {
+		result := This new(size)
+		result canvas draw(this, IntBox2D new(this size), IntBox2D new(size))
+		result
+	}
+	swapRedBlue: func {
+		this swapChannels(0, 2)
+	}
+	redBlueSwapped: func -> This {
+		result := this copy()
+		result swapRedBlue()
+		result
+	}
+	_createCanvas: override func -> Canvas { RasterRgbCanvas new(this) }
+
+	operator [] (x, y: Int) -> ColorRgb { this isValidIn(x, y) ? ((this buffer pointer + y * this stride) as ColorRgb* + x)@ : ColorRgb new(0, 0, 0) }
+	operator []= (x, y: Int, value: ColorRgb) { ((this buffer pointer + y * this stride) as ColorRgb* + x)@ = value }
+
+	open: static func (filename: String) -> This {
+		x, y, imageComponents: Int
+		requiredComponents := 3
+		data := StbImage load(filename, x&, y&, imageComponents&, requiredComponents)
+		buffer := ByteBuffer new(data as Byte*, x * y * requiredComponents, true)
+		This new(buffer, IntVector2D new(x, y))
```

Comment:
This function looks like the one in `RasterBgr`. I think STB loads the data as BGR and you will need to swap R<->B channels of the result.